### PR TITLE
Form Changes

### PR DESCRIPTION
**Pull request by eminis94**

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (cozej4) [inline .github/workflows/ci.yml:1]: Why was this file deleted? This is a useful configuration to make sure the app builds successful.
Please revert this change

File: .tx/config
Comment (cozej4) [inline .tx/config:1]: This files are important and are required. Do not delete them. Please reverse this change

File: opensrp-chw-hf/build.gradle
Comment (cozej4) [inline opensrp-chw-hf/build.gradle:1]: Reverse all changes made to build.gradle. Your changes should not touch anything on this file

File: opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_registration_form.json [hunk 1]
```diff
@@ -580,6 +586,42 @@
...
         }
       },
+      {
+        "key": "passport_number",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "person_attribute",
+        "openmrs_entity_id": "Passport",
+        "type": "mask_edit_text",
+        "mask": "TAE-###-###",
+        "mask_hint": "12345678912",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_registration_form.json]: This should be 123456, since only 6 digits are required from user input

File: opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_update_registration_info_form.json [hunk 2]
```diff
@@ -448,6 +454,42 @@
...
         }
       },
+      {
+        "key": "passport_number",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "person_attribute",
+        "openmrs_entity_id": "Passport",
+        "type": "mask_edit_text",
+        "mask": "TAE-###-###",
+        "mask_hint": "12345678912",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_update_registration_info_form.json]: This should be 123456, since only 6 digits are required

File: opensrp-chw-hf/src/main/assets/json.form/all_clients_registration_form.json [hunk 3]
```diff
@@ -580,6 +586,42 @@
...
         }
       },
+      {
+        "key": "passport_number",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "person_attribute",
+        "openmrs_entity_id": "Passport",
+        "type": "mask_edit_text",
+        "mask": "TAE-###-###",
+        "mask_hint": "12345678912",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/all_clients_registration_form.json]: This should be 123456

File: opensrp-chw-hf/src/main/assets/json.form/all_clients_update_registration_info_form.json [hunk 4]
```diff
@@ -448,6 +454,42 @@
...
         }
       },
+      {
+        "key": "passport_number",
+        "openmrs_entity_parent": "",
+        "openmrs_entity": "person_attribute",
+        "openmrs_entity_id": "Passport",
+        "type": "mask_edit_text",
+        "mask": "TAE-###-###",
+        "mask_hint": "12345678912",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/all_clients_update_registration_info_form.json]: This should be 123456

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 5]
```diff
@@ -99,11 +99,84 @@
...
         },
+        {
+          "name": "where_was_test_conducted",
+          "type": "radio_group",
+          "properties": {
+            "text": "Where was the test conducted?"
+          },
+          "meta_data": {
+            "openmrs_entity": "concept",
+            "openmrs_entity_id": "reason_no_action_was_taken",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: This is incorrect, the openmrs_entity_id should be "where_was_test_conducted" in this case

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 6]
```diff
@@ -99,11 +99,84 @@
...
+              "name": "other",
+              "text": "Other",
+              "meta_data": {
+                "openmrs_entity": "concept",
+                "openmrs_entity_id": "other",
+                "openmrs_entity_parent": ""
+              }
+            }
+          ],
+          "required_status": "yes:Please select action taken",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: This is should be, "yes:Please select where the test was conducted"

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 7]
```diff
@@ -138,6 +211,15 @@
...
               }
+            },
+            {
+              "name": "other",
+              "text": "Other",
+              "meta_data": {
+                "openmrs_entity": "concept",
+                "openmrs_entity_id": "reason_no_action_was_taken",
+                "openmrs_entity_parent": ""
+              }
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: This option is not placed in its correct position, the requirement was when others is selected in the question, "Where was the test conducted?" then add a field to capture other places where the test was conducted, but here you have added others in the question "Reasons client was not tested" which is not part of the requirement unless I have missed that requirement, please let me know

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 8]
```diff
@@ -146,6 +228,20 @@
...
+            "hint": "Others (Specify)"
+          },
+          "meta_data": {
+            "openmrs_entity": "concept",
+            "openmrs_entity_id": "reason_no_action_was_taken",
+            "openmrs_entity_parent": ""
+          },
+          "required_status": "true:Please specity other reason",
+          "subjects": "reason_no_action_was_taken:map, action_taken_hiv_problems:map"
+        },
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: Same here, the requirement is to add other places where the test was conducted, not other reasons no action was taken

File: opensrp-chw-hf/src/main/assets/rule/hiv_outcome_rules.yml [hunk 9]
```diff
@@ -6,13 +6,27 @@ condition: "action_taken_hiv_problems.containsKey('tested')"
...
 actions:
   - "reason_no_action_was_taken_visibility =  true"
 ---
+name: "other_reason_no_action_was_taken_visibility"
+description: "other_reason_no_action_was_taken visibility"
+priority: 1
+condition: "reason_no_action_was_taken.containsKey('other') && action_taken_hiv_problems.containsKey('no_action_taken')"
+actions:
+  - "other_reason_no_action_was_taken_visibility =  true"
+---
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/rule/hiv_outcome_rules.yml:29]: Same applies here, the added field should be other places where the test was conducted

**Review by cozej4 — CHANGES_REQUESTED** [review]

**Comment by cozej4** [comment]
Also add the issues in the project dashboard as we discussed in yesterday's meeting, and link the issues with your PR

File: opensrp-chw-hf/build.gradle
Comment (eminis94) [reply opensrp-chw-hf/build.gradle:1]: Noted. Can you please provide the username and password to login? I don't have the credentials for boresha-afya-stage url

File: opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_registration_form.json
Comment (eminis94) [inline opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_registration_form.json:1]: [Passport consist of only 6 numbers](https://github.com/Digital-Square-Tanzania/opensrp-client-chw-hf/issues/11)

File: opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_update_registration_info_form.json
Comment (eminis94) [inline opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_update_registration_info_form.json:1]: [Passport consists of only 6 numbers](https://github.com/Digital-Square-Tanzania/opensrp-client-chw-hf/issues/11)

File: opensrp-chw-hf/src/main/assets/json.form/all_clients_registration_form.json
Comment (eminis94) [inline opensrp-chw-hf/src/main/assets/json.form/all_clients_registration_form.json:1]: [Passport consists of only 6 numbers](https://github.com/Digital-Square-Tanzania/opensrp-client-chw-hf/issues/11)

File: opensrp-chw-hf/src/main/assets/json.form/all_clients_update_registration_info_form.json
Comment (eminis94) [inline opensrp-chw-hf/src/main/assets/json.form/all_clients_update_registration_info_form.json:1]: [Passport consists of only 6 numbers](https://github.com/Digital-Square-Tanzania/opensrp-client-chw-hf/issues/11)

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json
Comment (eminis94) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json:1]: [Add a question Where was the test conducted? ](https://github.com/Digital-Square-Tanzania/opensrp-client-chw-hf/issues/12)

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json (hunk 7 above)
Comment (eminis94) [reply opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: there are two scenarios where the other option appears the first one Reasons client was not tested" and add an option Others (Specify) and Where was the test conducted? - other option

File: opensrp-chw-hf/src/main/assets/rule/hiv_outcome_rules.yml (hunk 9 above)
Comment (eminis94) [reply opensrp-chw-hf/src/main/assets/rule/hiv_outcome_rules.yml:29]: there was a condition if "Tested" is selected then  Add a question after the Question "Action taken" - ( Where was the test conducted? )

File: opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_registration_form.json (hunk 1 above)
Comment (eminis94) [reply opensrp-chw-hf/src/main/assets/json.form-sw/all_clients_registration_form.json]: Noted

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 10]
```diff
@@ -99,11 +99,97 @@
...
+        {
+          "name": "other_where_was_test_conducted",
+          "type": "text_input_edit_text",
+          "properties": {
+            "hint": "Others (Specify)"
+          },
+          "meta_data": {
+            "openmrs_entity": "concept",
+            "openmrs_entity_id": "where_was_test_conducted",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: this should match the "name"

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 11]
```diff
@@ -138,6 +224,15 @@
                 "openmrs_entity_id": "test_not_required_based_on_risk_assessment",
                 "openmrs_entity_parent": ""
               }
+            },
+            {
+              "name": "other",
+              "text": "Other",
+              "meta_data": {
+                "openmrs_entity": "concept",
+                "openmrs_entity_id": "reason_no_action_was_taken",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: this should match the "name"

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json [hunk 12]
```diff
@@ -146,6 +241,20 @@
...
         },
+        {
+          "name": "other_reason_no_action_was_taken",
+          "type": "text_input_edit_text",
+          "properties": {
+            "hint": "Others (Specify)"
+          },
+          "meta_data": {
+            "openmrs_entity": "concept",
+            "openmrs_entity_id": "reason_no_action_was_taken",
```
Comment (cozej4) [inline opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: this should match the name

**Review by cozej4 — CHANGES_REQUESTED** [review]

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json (hunk 12 above)
Comment (eminis94) [reply opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: done

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json (hunk 11 above)
Comment (eminis94) [reply opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: done

File: opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json (hunk 10 above)
Comment (eminis94) [reply opensrp-chw-hf/src/main/assets/json.form/hiv_outcome.json]: done

**Review by cozej4 — APPROVED** [review]
